Joke Machine/Controller/ViewController.swift

```diff
@@ -49,21 +47,27 @@ class ViewController: UIViewController {
                 print("fail case")
             }
         }
-        (sender as AnyObject).setTitle("What's the answer?", for: .normal)
+        sender.setTitle("What's the answer?", for: .normal)
+        timer.invalidate()
         } else if (buttonLabel.titleLabel?.text == "What's the answer?") {
             punchlineLabel.isHidden = false
             punchlineLabel.text = jokes?.punchline
-            (sender as AnyObject).setTitle("Tell me a joke!", for: .normal)
+            sender.setTitle("Tell me a joke!", for: .normal)
             timer = Timer.scheduledTimer(timeInterval: 5.0, target: self, selector: #selector(fieldsClearer), userInfo: nil, repeats: false)
```

Comment:
her seferinde timerı oluşturmak gereksiz memory kullandırır. Tepede oluşturup işin bittiğinde invalidate edebilirsin